src/java/org/apache/cassandra/index/sai/view/IndexViewManager.java

```diff
@@ -186,20 +188,20 @@ public void prepareSSTablesForRebuild(Collection<SSTableReader> sstablesToRebuil
      */
     public void invalidate(boolean indexWasDropped)
     {
-        View oldView, newView;
+        View oldView, newView = null;
         do
         {
-            oldView = view.get();
+            // We skip referencing because we do not use its indexes.
+            oldView = viewRef.get();
+            if (newView != null)
+                newView.release();
+            // TODO are we non queryable at this point?
```

Comment:
at this point, `Index` is removed from `SecondaryIndexManager`. There might be in-flight query but no more new query starting